src/components/MonthCalendar/index.js

```diff
@@ -0,0 +1,170 @@
+import React, { useMemo } from 'react';
+import PropTypes from 'prop-types';
+import { useUniqueIdentifier, useLocale } from '../../libs/hooks';
+import { getFormattedMonth, getYearsRange, getLastDayMonth, addMonths } from '../Calendar/helpers';
+import getCurrentMonth from './helpers/getCurrentMonth';
+import { Provider } from './context';
+import Select from '../Select';
+import RightIcon from './icons/rightArrow';
+import LeftIcon from './icons/leftArrow';
+import DaysOfWeek from './daysOfWeek';
+import Month from './month';
+import StyledControlsContainer from './styled/controlsContainer';
+import StyledMonthContainer from './styled/monthContainer';
+import StyledArrowButton from './styled/arrowButton';
+import StyledMonth from './styled/month';
+import StyledTable from './styled/table';
+
+export default function MonthCalendar(props) {
+    const {
+        id,
+        className,
+        style,
+        minDate,
+        maxDate,
+        currentMonth,
+        selectedDate,
+        locale,
+        onSelectDate,
+        onMonthChanged,
+        dateComponent,
+    } = props;
+    const monthLabelId = useUniqueIdentifier('month');
+    const currentLocale = useLocale(locale);
+    const month = useMemo(() => getCurrentMonth(currentMonth, minDate, maxDate), [
+        currentMonth,
+        minDate,
+        maxDate,
+    ]);
+    const currentYear = month.getFullYear();
+    const yearsRange = useMemo(
+        () =>
+            getYearsRange({
+                minDate,
+                maxDate,
+                currentMonth: month.getMonth(),
+            }),
+        [minDate, maxDate, month],
+    );
+    const formattedMonth = useMemo(() => getFormattedMonth(month, currentLocale), [
+        currentLocale,
+        month,
+    ]);
+
+    const lastYearItem = yearsRange[yearsRange.length - 1];
+    const maxSelectableDate = maxDate || new Date(lastYearItem.value, 11, 31);
+    const disableNextMonth = addMonths(month, 1) > maxSelectableDate;
+    const minSelectableDate = minDate || new Date(yearsRange[0].value, 0, 1);
+    const prevDate = useMemo(() => getLastDayMonth(addMonths(month, -1)), [month]);
+    const disablePreviousMonth = prevDate < minSelectableDate;
+
+    const selectPreviousMonth = () => {
+        const newMonth = addMonths(month, -1);
+        newMonth.setHours(0, 0, 0, 0);
+        return onMonthChanged(newMonth);
```

Comment:
because event are part of the API we should be very careful ... it's better to pass an event object that scale if we need to send more data later
```js
return onMonthChanged({ newMonth });
```

also I will call it `month` instead of newMonth